pkg/cache/multi_namespace_cache.go

```diff
@@ -145,28 +164,40 @@ func (c *multiNamespaceCache) List(ctx context.Context, list runtime.Object, opt
 		return cache.List(ctx, list, opts...)
 	}
 
-	// Get all the objects in the namespaces we are watching.
-	gvk, err := apiutil.GVKForObject(list, c.Scheme)
+	listAccessor, ok := list.(metav1.ListInterface)
```

Comment:
It seems you don't have to do the type assertion yourself.
You can use https://github.com/kubernetes-sigs/controller-runtime/blob/e2d83e883d17206248db563cbc71f023d0344d3b/vendor/k8s.io/apimachinery/pkg/api/meta/meta.go#L69-L93.